test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -420,22 +420,32 @@ class SystemInfo:
     """
     logger = logging.getLogger(__name__)
 
-    def __init__(self, hostname=None):
+    def get_system_info(self, hostname=None):
+        """
+        used to get system's ram size and disk size information.
+
+        :system_ram: Available ram(in GB) of the test running machine
+        :system_disk: Available disk size(in GB) of the test running machine
+        """
         du = DshUtils()
         # getting RAM size in gb
         mem_info = du.cat(hostname, "/proc/meminfo")
         if mem_info['rc'] != 0:
-            self.logger.error('failed to get content of /proc/meminfo')
+            _msg = 'failed to get content of /proc/meminfo of host: '
+            self.logger.error(_msg + hostname)
         else:
-            ram_info = dict((i.split()[0].rstrip(':'), int(i.split()[1]))
-                            for i in mem_info['out'])
-            self.system_ram = float(ram_info['MemAvailable']) / (2**20)
+            ram_dict = {}
+            for i in mem_info['out']:
+                ram_info = i.split()
```

Comment:
why not just split once at the colon? right now, it will split as many times as the spaces in the string, which might give you unexpected results. The following might be safer:
key, val = i.split(":", 1)
ram_dict[key.strip()] = val.strip()

Reply:
Actually, why are you using ram_dict at all? All that you need is MemAvailable right? So you can just do this:

```
for i in mem_info['out']:
    if "MemAvailable" in i:
        self.system_ram = float(i.split(":", 1)[1].strip()) / (2**20)
        break
```

Reply:
Here i is "MemAvailable: 1234 kb".i think split by space is nice.